packages/firebase_auth/lib/firebase_auth.dart

```diff
@@ -73,12 +73,31 @@ class FirebaseAuth {
     'plugins.flutter.io/firebase_auth',
   );
 
+  final List<StreamController<FirebaseUser>> _authStateChangedController =
```

Comment:
I would use a `Map<int, StreamController<FirebaseUser>>` instead of a `List` because `List` isn't sparse and you can have a potentially unbounded number of controllers created over time.